### ISSUE TITLE
Discussion about native `blocks` implementation 

### ISSUE THREAD

**Issue by mhalano** [GothenburgBitFactory/taskwarrior#2608]
This issue is to continue the discuss in #2148 about a `blocks` (opposite of `depends`) implemented natively and not using hooks.

**Comment by mhalano** [comment]
There are the obvious limitation of when you are using the hook you can't remove the blocks option.

**Comment by dark-ether** [comment]
it would be possible to implement with hooks but currently there isn't any active project implementing a blocks attribute. and at least for my workflow it would be really helpful

**Comment by willhansen** [comment]
Related discussion: https://github.com/GothenburgBitFactory/taskwarrior/discussions/2528
- The linked hook: https://gist.github.com/wbsch/a2f7264c6302918dfb30

Surprised me that this isn't built in.  There's the classic "yak shaving" use case where you're doing a task, but wait you need to do another task first. But wait, there's another one before that.  etc, etc.

**Comment by Screwtapello** [comment]
It's *mostly* possible to have a `blocks` field with a hook that tries to mirror the property to the `depends` field and back, as the linked hook does. The basic principle is: when task X gets task Y added to its `blocks` field, run `task Y modify depends:X`.

The first problem is that each invocation of TaskWarrior updates the database independently, so if the hook *also* launches TaskWarrior, the outer TaskWarrior can accidentally undo changes made by the inner one. I'm not exactly sure what changes can be lost this way, but this situation is described in more detail in https://github.com/GothenburgBitFactory/taskwarrior/issues/1726#issuecomment-365653573

As an alternative to running TaskWarrior directly, the above linked hook writes a log of pending updates that get applied by an on-launch hook next time TaskWarrior is run. Along with the shell-quoting issues this specific implementation is vulnerable to, I don't like the idea of some of my database state living in a poorly-documented extra file for an unbounded amount of time — if I back up the TaskWarrior database and forget to back up this extra file, or forget to set up the hook after restoring the backup, my database is not in the state I thought it was in.

Whether invoked directly or at some later point, the two-step process has the additional problem of breaking `task undo`, since a single user-facing operation becomes multiple database operations, and you have to `undo` them all in order to completely undo the change. I don't think it's possible for the hook to detect that this is happening, and remind the user to undo an additional time.

TaskWarrior has a lot of convenience features around the `depends` field that cannot easily be replicated in a hook. For example, you can say `task 1 modify depends:2,4-6` and to depend on tasks 2, 4, 5, and 6. Except tasks can be renumbered, so it doesn't actually store those ID numbers but stores the tasks' UUIDs. One can resolve a list of IDs to UUIDs with the `_uuids` command, but it behaves differently: `modify depends:1,2,999` will fail with an error that task 999 doesn't exist, but `1,2,999 _uuids` will just print two UUIDs and report no error. Despite all the advice that tools should use `import` and `export` rather than `modify`, I don't think it's possible to reliably implement a `blocks` hook without it.

In addition, while `task 1 modify depends:1` will detect the circular dependency and fail, `task import` does not implement the same consistency checks, so the hook must use `modify` to get correct results.
